### PR TITLE
Fixed setting of _save_dir when run initiated externally

### LINKED ISSUE #7103: Trainer crashes on checkpoint save when wandb run is initiated externally and passed into the WandbLogger
When defining the wandb run externally and passing it into the WandbLogger, the `_save_dir` property of the model is not set.

I've made an update to the WandbLogger to accommodate for the change in the pull request attached to this issue.

image

https://github.com/PyTorchLightning/pytorch-lightning/blob/6b15ca95f07d5499e470d5027699770b3c2b95fd/pytorch_lightning/loggers/wandb.py#L158-L178

### PR DESCRIPTION
**Pull request by THasthika**
## What does this PR do?

It moves the setting of the `_save_dir` of WandbLogger from the clause where it only runs when the experiment is set to `None`.

With this change we can set the experiment externally and have the models be uploaded to the wandb servers.

Fixes #7103

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by awaelchli** [comment]
Could you add/modify the tests in `tests/loggers/test_wandb.py` or show us a reproducible example so we can help do that. Thanks for sending a fix <3

**Comment by THasthika** [comment]
> Could you add/modify the tests in `tests/loggers/test_wandb.py` or show us a reproducible example so we can help do that. Thanks for sending a fix <3

I've added a single test to check if the `save_dir` is set in the logger when experiment is set manually. The test is in the `test_wandb_logger_init` function in `tests/loggers/test_wandb.py`. Will that be alright?

**Review by awaelchli — APPROVED** [review]

**Comment by awaelchli** [comment]
Well done

File: tests/loggers/test_wandb.py [hunk 1]
```diff
@@ -42,6 +42,15 @@ def test_wandb_logger_init(wandb, recwarn):
     wandb.init.assert_called_once()
     wandb.init().log.assert_called_once_with({'acc': 1.0})
 
+    # test wandb.init and setting logger experiment externally
+    wandb.run = None
+    run = wandb.init()
+    logger = WandbLogger(experiment=run)
+    assert logger.experiment
+    assert run.dir is not None
+    assert logger.save_dir is not None
```
Comment (carmocca) [inline tests/loggers/test_wandb.py:51]: redundant assert

```suggestion
```

**Review by carmocca — APPROVED** [review]

**Pull request merged**